### ISSUE TITLE
Nested `CREATE` with broadcast from custom sender fails: `tx.origin` becomes wrong, and nonce does not increment

### ISSUE THREAD

**Issue by adhusson** [foundry-rs/foundry#5062]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (33f3fee 2023-05-27T00:09:05.418420000Z)

### What command(s) is the bug in?

forge script

### Operating System

macOS (Apple Silicon)

### Describe the bug

The following script fails. It should succeed.

```solidity
pragma solidity ^0.8.17;

import {Script, console} from "forge-std/Script.sol";

contract Contract {}
contract SubContract {}
contract BadContract {
  constructor() {
    new SubContract();
  }
}
contract NestedCreateFail is Script {
  function run() public {
    address sender = address(uint160(uint(keccak256("woops"))));

    vm.broadcast(sender);
    new BadContract();

    // at this point: sender's nonce is 0 (should be 1), tx.origin == sender (should be DEFAULT_SENDER)
    
    vm.broadcast(sender);
    new Contract();
  }
}
```

The error is just ` "EvmError: Revert"`. I believe it is due to attempting to deploy a contract at an address that has nonzero code.

This occurs since 0f98eef9a040468a1d01c57b800565aeba805166.

Also, if you do `vm.setNonce(sender,2)` before deploying `Contract`, the script works. But the script does _not_ work if you instead write `vm.setNonce(sender,1)`, unless you precede that with `vm.getNonce(sender)`. This could be an unrelated bug.

**Comment by mattsse** [comment]
ptal @Evalir 
this seems to be a widespread issue

probably introduced in https://github.com/foundry-rs/foundry/pull/4469

**Comment by Evalir** [comment]
Fixed with #5100
